### PR TITLE
Employer APIs without Unit Test

### PR DESCRIPTION
**Pull request by pinkball89**
I will try to finish the Unit Test later. Please review and give me any feedback, I really appreciate your time.

### CODE REVIEW

File: src/main/java/vn/unigap/api/controller/EmployerController.java [hunk 1]
```diff
@@ -98,4 +100,15 @@ public ResponseEntity<ApiResponseDTO> deleteEmployer(@PathVariable long id) {
         }
 
     }
+
+    private EmployerResponseDTO mapToResponseDTO(Employer employer) {
```
Comment (phungbuuquang) [inline src/main/java/vn/unigap/api/controller/EmployerController.java:93]: Should not put it in controller. You should create a class named with suffix: DtoOut and put this function in it.

File: src/main/java/vn/unigap/api/entity/Employer.java [hunk 2]
```diff
@@ -60,14 +66,18 @@ public void setName(String name) {
...
     }
 
-    public int getProvince() {
-        return province;
+    public int getProvinceId() {
+        return province.getId();
     }
 
-    public void setProvince(int province) {
+    public void setProvince(Province province) {
```
Comment (phungbuuquang) [inline src/main/java/vn/unigap/api/entity/Employer.java]: you can use @Data annotation on top class , you will not declare function setter, getter.

Comment (pinkball89) [reply src/main/java/vn/unigap/api/entity/Employer.java]: Thanks.

File: src/main/java/vn/unigap/api/controller/EmployerController.java (hunk 1 above)
Comment (pinkball89) [reply src/main/java/vn/unigap/api/controller/EmployerController.java:93]: I do not want to expose EmployerResponseDTO to Employer so I put this function under EmployerController as EmployerController  can be exposed to both Employer and EmployerResponseDTO.

This is just the coding style, when the project is more complicated (dealing with decoupling) then we can think about where we should put it later.

File: src/main/java/vn/unigap/api/entity/Employer.java (hunk 2 above)
Comment (pinkball89) [reply src/main/java/vn/unigap/api/entity/Employer.java]: I did change it. Please help me to review! TIA!

File: pom.xml [hunk 3]
```diff
@@ -33,8 +33,13 @@
 		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+			<version>3.1.11</version>
```
Comment (robert28893) [inline pom.xml:37]: No need explicit version for this dependency

File: src/main/java/vn/unigap/api/service/EmployerService.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.*;
+
+
+
+@Service
+public class EmployerService {
+    private final ProvinceRepository provinceRepository;
+    private final EmployerRepository employerRepository;
+
+    public static class BadInputException extends RuntimeException {
```
Comment (robert28893) [inline src/main/java/vn/unigap/api/service/EmployerService.java]: you should declare these exception classes to external files for common usages

File: src/main/java/vn/unigap/api/controller/EmployerController.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            Employer employer = employerService.createEmployer(requestCreateDTO);
+            return ResponseEntity.status(HttpStatus.CREATED)
+                    .body(new ApiResponseDTO(0, 201, "Employer created successfully", employer));
+
+        } catch (Exception e) {
+            return toApiResponse(e);
+        }
+    }
+
+    private static ResponseEntity<ApiResponseDTO> toApiResponse(Exception e) {
```
Comment (robert28893) [inline src/main/java/vn/unigap/api/controller/EmployerController.java]: Should consider to move this function to a common class. May be is a abstract class or a common util class

Comment (pinkball89) [reply src/main/java/vn/unigap/api/controller/EmployerController.java]: Done. Move to api/exception/Exceptionhandler.java

File: src/main/java/vn/unigap/api/service/EmployerService.java (hunk 4 above)
Comment (pinkball89) [reply src/main/java/vn/unigap/api/service/EmployerService.java]: Done. Move to api/exception/ApiException.java

**Review by robert28893 — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (pinkball89) [reply pom.xml:37]: Noted coach.

**Pull request merged**